server/controllers/admin/holidays.js

```diff
@@ -0,0 +1,147 @@
+/**
+* Holiday Controller
+*
+* This controller exposes an API to the client for reading and writing Holiday
+*/
+
+var db = require('../../lib/db');
+var NotFound = require('../../lib/errors/NotFound');
+var Unauthorized = require('../../lib/errors/Unauthorized');
+
+// GET /Holiday
+function lookupHoliday(id) {
+  var sql =`
+    SELECT h.id, h.label, h.employee_id, h.dateFrom, h.dateTo, h.percentage
+    FROM holiday AS h  
+    WHERE h.id = ?`;
+
+  return db.one(sql, [id]);
+}
+
+// Check Holidays
+/** 
+*
+* This function prevents to define for an employee, two periods of holidays that fits nested
+*/
+function checkHoliday(param) {
+  var sql = `
+    SELECT id, employee_id, label, dateTo, percentage, dateFrom 
+    FROM holiday WHERE employee_id = ?
+    AND ((dateFrom >= DATE(?)) OR (dateTo >= DATE(?)) OR (dateFrom >= DATE(?))
+    OR (dateTo >= DATE(?)))
+    AND ((dateFrom <= DATE(?)) OR (dateTo <= DATE(?)) OR (dateFrom <= DATE(?))
+    OR (dateTo <= DATE(?)))
+  `;
+
+  return db.exec(sql, [param.employee_id, param.dateFrom, param.dateFrom, param.dateTo, param.dateTo, param.dateFrom, param.dateFrom, param.dateTo, param.dateTo]);
+}
+
+
+// Lists the Payroll Holidays
+function list(req, res, next) {
+  const sql = `
+    SELECT h.id, h.label, h.employee_id, h.dateFrom, h.dateTo, p.display_name, h.percentage
+    FROM holiday AS h
+    JOIN employee AS e ON e.id = h.employee_id
+    JOIN patient AS p ON p.uuid = e.patient_uuid
+  ;`;
+
+  db.exec(sql)
+    .then((rows) => {
+      res.status(200).json(rows);
+    })
+    .catch(next)
+    .done();
+}
+
+/**
+* GET /Holiday/:ID
+*
+* Returns the detail of a single Holiday
+*/
+function detail(req, res, next) {
+  var id = req.params.id;
+
+  lookupHoliday(id)
+    .then((record) => {
+      res.status(200).json(record);
+    })
+    .catch(next)
+    .done();
+}
+
+// POST /Holiday
+function create(req, res, next) {
+  const sql = `INSERT INTO holiday SET ?`;
+  const data = req.body;
+
+  checkHoliday(data)
+    .then((record) => {
+      if (record.length) {
+        throw new Unauthorized('Holiday Nested.', 'ERRORS.HOLIDAY_NESTED');
```

Comment:
I don't think this should be unauthorized (403).  Probably a `BadRequest` error is better (400).